### DISCUSSION TITLE
Handling window and localStorage in SSR

### DISCUSSION THREAD

**Discussion by erfanmola** [inertiajs/inertia#1638] [Help]
Hi, I'm sorry if this is a newbie question, but How can I handle a code like this when using SSR:
`window.addEventListener();` or `localStorage.set()` ?

This code is working fine in client side, but crashes when used in Server side, what is the correct way of handling these kind of things? I'm using Vue3 and Laravel, My application flow depends on these events so I cannot just delete it, or check if window is defined or ...

**Comment by erfanmola** [comment]
I found the solution, codes like window.xxx or localStorage and etc, must be used inside functions like onMounted, onBeforeMount, ...

**Reply by Uzeyir9695** [reply]
Thank you. This solved my issue.
